src/main/java/com/example/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.example.store.service.impl;
+
+import com.example.store.dto.cartitem.CartItemRequestDto;
+import com.example.store.dto.shoppingcart.ShoppingCartDto;
+import com.example.store.exception.EntityNotFoundException;
+import com.example.store.mapper.ShoppingCartMapper;
+import com.example.store.model.ShoppingCart;
+import com.example.store.model.User;
+import com.example.store.repository.shoppingcart.ShoppingCartRepository;
+import com.example.store.repository.user.UserRepository;
+import com.example.store.service.CartItemService;
+import com.example.store.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemService cartItemService;
+    private final UserRepository userRepository;
+
+    @Override
+    public ShoppingCartDto registerNewShoppingCart(User user) {
+        return shoppingCartMapper.toDto(createShoppingCart(user));
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addItemToShoppingCart(CartItemRequestDto requestDto, String email) {
+        ShoppingCart shoppingCart = getShoppingCart(email);
+        shoppingCart.getCartItems().add(cartItemService.save(requestDto, shoppingCart));
```

Comment:
i'm not sure what you are doing here, but it looks inefficient. Are you sure this is not redundant?